### DISCUSSION TITLE
Compiling static binary unix

### DISCUSSION THREAD

**Discussion by takov751** [micropython/micropython#12339] [Unix & Windows]
What is a recommended way to compile unix binary which than could be used on different distros? 

My initial idea behind this.

- Creating a docker from scratch running only micropython binary in it.
- creating a static build for embedded devices , especially old models.

As a case study i started with mips architecture. I was able to create binary with dockcross

**Comment by andrewleech** [comment]
There's already a Unix build in docker here: https://hub.docker.com/r/micropython/unix
It's not yet built automatically though, @mattytrentini is maintaining it.

Prior to GitHub actions being used here I maintained a Gitlab CI system to build all the ports including Unix.
Build script: https://gitlab.com/alelec/micropython_ci/-/blob/master/ports/unix/.gitlab-ci.yml

To build the unix port I use a toolchain in a very old os docker image to link against old version libraries to have as broad compatibility as possible: https://gitlab.com/alelec/docker-oldlinux-gcc

**Reply by takov751** [reply]
Indeed i was following in the footsteps of that build, however when it comes to odd embedded devices even the old builder env can  causes some issues for me. Nevertheless thank you i will look into it
